<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>
    To avoid draining the battery, an Android device that is left idle quickly falls asleep.
    Hence, keeping the screen on should be avoided, unless it is absolutely necessary.
    If so, developers typically use the `FLAG_KEEP_SCREEN_ON` in their activity.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
</pre>
